jira_test.go

```diff
@@ -700,3 +701,133 @@ func TestOpenJiraTicketDryRyn(t *testing.T) {
 	return
 
 }
+
+/// test read config file
+func TestParseConfigFileFunc(t *testing.T) {
+
+	assert := assert.New(t)
+
+	// Moving the fixture to root (where we expect the jira.yaml fiel to be)
+	err := os.Rename("./fixtures/jiraConfig.yaml", "./jira.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	config := parseConfigFile()
+
+	// Moving the config file back
+	err = os.Rename("./jira.yaml", "./fixtures/jiraConfig.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	assert.Equal(config.Snyk.OrgID, "a1b2c3de-99b1-4f3f-bfdb-6ee4b4990513")
+	assert.Equal(config.Snyk.Severity, "critical")
+	assert.Equal(config.Snyk.MaturityFilter, "mature")
+	assert.Equal(config.Snyk.IssueType, "all")
+	assert.Equal(config.Snyk.PriorityScoreThreshold, 10)
+	assert.Equal(config.Snyk.RemoteUrl, "github.com/owner/repo")
+	assert.Equal(config.Snyk.EndpointAPI, "")
+
+	assert.Equal(config.Jira.JiraTicketType, "Task")
+	assert.Equal(config.Jira.JiraProjectID, "12345")
+	assert.Equal(config.Jira.AssigneeId, "123abc456def789")
+	assert.Equal(config.Jira.AssigneeName, "AccountName")
+	assert.Equal(config.Jira.PriorityIsSeverity, true)
+
+}
+
+/// test setting flags
+func TestSetOptionFunc(t *testing.T) {
+
+	assert := assert.New(t)
+
+	// reset command line arg
+	oldArgs := os.Args
+	defer func() { os.Args = oldArgs }()
+
+	// Moving the fixture to root (where we expect the jira.yaml fiel to be)
+	err := os.Rename("./fixtures/jiraConfigV2.yaml", "./jira.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Args = append(os.Args, "-token=123")
+	os.Args = append(os.Args, "-configFile=true")
+
+	options := flags{}
+	options.setOption()
+
+	// Moving the config file back
+	err = os.Rename("./jira.yaml", "./fixtures/jiraConfigV2.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	assert.Equal(options.mandatoryFlags.orgID, "456")
```

Comment:
I'd suggest to have a logic to test all flags in each mandatory and optional flags structs. Makes it easier to then just focus on updating the fixture(s)